components/sup/src/manager/mod.rs

```diff
@@ -823,6 +824,50 @@ impl Manager {
         ))
     }
 
+    pub fn service_group_cfg(
+        mgr: &ManagerState,
+        req: &mut CtlRequest,
+        opts: protocol::ctl::SvcGetCfg,
+    ) -> NetResult<()> {
+        let service_group: ServiceGroup = opts.service_group.ok_or(err_update_client())?.into();
+        let cfg_type = opts
+            .cfg_type
+            .and_then(protocol::types::ServiceCfgType::from_i32)
+            .unwrap_or_default();
+        let mut msg = protocol::types::ServiceCfg {
+            format: Some(protocol::types::service_cfg::Format::Toml as i32),
+            default: None,
+        };
+        for service in mgr.services.read().unwrap().iter() {
+            if service.service_group == service_group {
+                let tomlcfg = match cfg_type {
+                    ServiceCfgType::Gossip => toml::to_string_pretty(&toml::value::Value::Table(
+                        service.cfg.gossip.clone().unwrap_or_default(),
+                    )),
+                    ServiceCfgType::Default => toml::to_string_pretty(&toml::value::Value::Table(
+                        service.cfg.default.clone().unwrap_or_default(),
+                    )),
+                    ServiceCfgType::User => toml::to_string_pretty(&toml::value::Value::Table(
+                        service.cfg.user.clone().unwrap_or_default(),
+                    )),
+                    ServiceCfgType::Environment => {
+                        toml::to_string_pretty(&toml::value::Value::Table(
+                            service.cfg.environment.clone().unwrap_or_default(),
+                        ))
+                    }
+                    ServiceCfgType::Merged => toml::to_string_pretty(&service.cfg),
+                };
+                msg.default = Some(tomlcfg.unwrap_or("".to_string()));
```

Comment:
I think if we get an error here, then that means that there was an error serializing things to a pretty-printed string, in which case it'd be nice to not swallow that error (if indeed it is even possible... I'd like to think that by the time we get here, any configuration would be at least valid TOML).